<launch>
  <arg name="map_size_x_"/>
  <arg name="map_size_y_"/>
  <arg name="map_size_z_"/>

 <arg name="odometry_topic"/>
 <arg name="sensor_pose_topic"/>
  <arg name="depth_topic"/>
 <arg name="semantic_topic"/>
 <arg name="cloud_topic"/>

  <arg name="cx"/>
  <arg name="cy"/>
  <arg name="fx"/>
  <arg name="fy"/>

  <!-- main node -->
  <node pkg="plan_env" name="test_sdf" type="test_sdf" output="screen">
   <remap from ="/odom_world" to="$(arg odometry_topic)"/>
   <remap from ="/map_ros/pose"   to = "$(arg sensor_pose_topic)"/>
    <remap from ="/map_ros/depth" to = "$(arg depth_topic)"/>
   <remap from ="/map_ros/semantics" to= "$(arg semantic_topic)"/>
    <remap from ="/map_ros/cloud" to="$(arg cloud_topic)"/>

   <param name="map_ros/do_semantics" value="false" type="bool"/>
   <param name="map_ros/input_rdf" value="true" type="bool"/>
   <param name="map_ros/image_rows" value="480"/>
   <param name="map_ros/image_cols" value="640"/>

    <param name="sdf_map/resolution"      value="0.1" />
    <param name="sdf_map/map_size_x"   value="$(arg map_size_x_)" /> 
    <param name="sdf_map/map_size_y"   value="$(arg map_size_y_)" /> 
    <param name="sdf_map/map_size_z"   value="$(arg map_size_z_)" /> 
    <param name="sdf_map/obstacles_inflation"     value="0.0" />
   <param name="sdf_map/person_obstacles_inflation"     value="0.0" />
    <param name="sdf_map/local_bound_inflate"    value="0.5"/>
    <param name="sdf_map/local_map_margin" value="50"/>
    <param name="sdf_map/ground_height"        value="-1.0"/>
    <param name="sdf_map/default_dist"        value="0.0"/>
    
    <param name="sdf_map/p_hit"  value="0.70"/>
    <param name="sdf_map/p_miss" value="0.35"/>
    <param name="sdf_map/p_min"  value="0.12"/>
    <param name="sdf_map/p_max"  value="0.97"/>
    <param name="sdf_map/p_occ"  value="0.80"/>
    <param name="sdf_map/min_ray_length" value="0.5"/>
    <param name="sdf_map/max_ray_length" value="5"/>
    <param name="sdf_map/virtual_ceil_height"   value="-10"/>
    <param name="sdf_map/optimistic" value="false" type="bool"/>
    <param name="sdf_map/signed_dist" value="false" type="bool"/>
   <param name="/map_ros/do_semantics"      value="false" type="bool" />

    <param name="map_ros/cx" value="$(arg cx)"/>
    <param name="map_ros/cy" value="$(arg cy)"/>
    <param name="map_ros/fx" value="$(arg fx)"/>
    <param name="map_ros/fy" value="$(arg fy)"/>
    <param name="map_ros/depth_filter_maxdist"   value="10.0"/>
    <param name="map_ros/depth_filter_mindist"   value="0.1"/>
    <param name="map_ros/depth_filter_margin"    value="0"/>
    <param name="map_ros/k_depth_scaling_factor" value="1000.0"/>
    <param name="map_ros/skip_pixel" value="2"/>
    <param name="map_ros/esdf_slice_height" value="0.3"/>
    <param name="map_ros/visualization_truncate_height"   value="10.09"/>
    <param name="map_ros/visualization_truncate_low"   value="-2.0"/>
    <param name="map_ros/show_occ_time"  value="false"/>
    <param name="map_ros/show_esdf_time" value="false"/>
    <param name="map_ros/show_all_map" value="true"/>
    <param name="map_ros/frame_id"      value="world"/>
  </node>

</launch>